qiskit/providers/__init__.py

```diff
@@ -500,6 +546,110 @@ def result(self):
         def status(self):
             return JobStatus.DONE
 
+======================================
+Migrating between Backend API Versions
+======================================
+
+BackendV1 -> BackendV2
+======================
+
+The :obj:`~BackendV2` class re-defined user access for most properties of a
+backend to make them work with native Qiskit data structures and have flatter
+access patterns. However this means when using a provider that upgrades
+from :obj:`~BackendV1` to :obj:`~BackendV2` existing access patterns will need
+to be adjusted. It is expected for existing providers to deprecate the old
+access where possible to provide a graceful migration, but eventually users
+will need to adjust code. The biggest change to adapt to in :obj:`~BackendV2` is
+that most of the information accesible about a backend is contained in it's
+:class:`~qiskit.transpiler.Target` object and the backend's attributes often query
+it's :attr:`~qiskit.providers.BackendV2.target`
+attribute to return information, however in many cases the attributes only provide
+a subset of information the target can contain. For example, ``backend.coupling_map``
+returns a :class:`~qiskit.transpiler.CouplingMap` constructed from the
+:class:`~qiskit.transpiler.Target` accesible in the
+:attr:`~qiskit.providers.BackendV2.target` attribute, however the target may contain
+instructions that operate on more than two qubits (which can't be represented in a
+:class:`~qiskit.transpiler.CouplingMap`) or has instructions that only operate on
+a subset of qubits (or two qubit links for a two qubit instruction) which won't be
+detailed in the full coupling map returned by
+:attr:`~qiskit.providers.backend_v2.coupling_map`. So depending on your use case
+it might be necessary to look deeper than just the equivalent access with
+:obj:`~BackendV2`.
+
+Below is a table of example access patterns in :obj:`~BackendV1` and the new form
+with :obj:`~BackendV2`:
+
+.. list-table:: Migrate from :obj:`~BackendV1` to :obj:`~BackendV2`
+   :header-rows: 1
+
+   * - :obj:`~BackendV1`
+     - :obj:`~BackendV2`
+     - Notes
+   * - ``backend.configuration().n_qubits``
+     - ``backend.num_qubits``
+     -
+   * - ``backend.configuration().coupling_map``
+     - ``backend.coupling_map``
+     - The return from :obj:`~BackendV2` is a :class:`~qiskit.transpiler.CouplingMap` object.
+       while in :obj:`~BackendV1` it is jsut an edge list. Also this is just a view of
+       the information contained in ``backend.target`` which may only be a subset of the
+       information contained in :class:`~qiskit.transpiler.Target` object.
```

Comment:
I just noticed it now - nothing is generating the `Target` doc stubs, so all the `Target` crossrefs fail.  I imagine it needs an `autoclass` or an `autosummary` in the `transpiler` docstring (probably in the other PR).

Reply:
Yeah good catch. It's missing from the autodoc in `transpiler/__init__.py` in #5885. It's re-exported from `__init__.py` but for some reason I didn't add it to an autosummary directive. I'll update that now over there and then rebase this after

Reply:
Ok, I updated #5885 to include `Target` in the docs and rebased this now on top of that so the crossrefs should work now.